source/common/config/utility.h

```diff
@@ -495,6 +497,21 @@ class Utility {
     }
     return std::make_unique<FixedBackOffStrategy>(dns_refresh_rate_ms);
   }
+
+  /**
+   * Return the hash_key value inside the envoy.lb metadata filter value.
+   * @param metadata the metadata filters.
+   * @throws EnvoyException if there the value is not empty or a string type.
+   */
+  static std::string getHashKeyMetadataValue(const envoy::config::core::v3::Metadata* metadata) {
+    const ProtobufWkt::Value& val = Metadata::metadataValue(
+        metadata, MetadataFilters::get().ENVOY_LB, MetadataEnvoyLbKeys::get().HASH_KEY);
+    if (val.kind_case() != val.kStringValue && val.kind_case() != val.KIND_NOT_SET) {
+      ExceptionUtil::throwEnvoyException(
+          fmt::format("hash_key must be string type, got: {}", val.kind_case()));
+    }
```

Comment:
We can't throw exceptions on the data path. I think fail back is probably fine, with a debug log? Please make sure you have a test for this case also.